### PR TITLE
Update Android Toolchain

### LINKED ISSUE #22: Android: migrate from deprecated `android` tool
```
$ android
*************************************************************************
The "android" command is deprecated.
For manual SDK, AVD, and project management, please use Android Studio.
For command-line tools, use tools/bin/sdkmanager and tools/bin/avdmanager
*************************************************************************
```

### PR DESCRIPTION
**Pull request by Ghabry**
This should be okay but the Player gives linker errors.

Fix #22

### CODE REVIEW

**Comment by Ghabry** [comment]
I guess the libm in Wildmidi can be removed in general. Because in Player I don't check for libm and CMake still adds it to the linker line :)

**Comment by carstene1ns** [comment]
Well, libm is part of the c lib on some platforms, but some devs and build systems are clever and do not mind overlinking it.

**Comment by carstene1ns** [comment]
[glibc 2.26 has dropped `xlocale.h`](https://sourceware.org/glibc/wiki/Release/2.26#Removal_of_.27xlocale.h.27), so we need to patch `icu-native/source/i18n/digitlst.cpp`.

File: android/2_build_toolchain.sh [hunk 1]
```diff
@@ -66,24 +78,33 @@ function install_lib {
...
+	export CPPFLAGS=$CPPFLAGSOLD
+}
+
 # Install SDL2
 function install_lib_sdl {
 	# $1 => platform (armeabi armeabi-v7a x86 mips)
 
-	cd SDL2-2.0.4
+	cd SDL2-2.0.6
 	echo "APP_STL := gnustl_static" > "jni/Application.mk"
```
Comment (carstene1ns) [inline android/2_build_toolchain.sh]: I guess this should be changed to `c++_static`.

Comment (Ghabry) [reply android/2_build_toolchain.sh]: Good find. Hmm, guess the only reason this didn't crash is the lack of c++ in SDL

Comment (carstene1ns) [reply android/2_build_toolchain.sh]: yep.

Comment (Ghabry) [reply android/2_build_toolchain.sh]: Yepp, should default to clang by now.

**Comment by carstene1ns** [comment]
The liblcf buildscript uses an outdated path for `$NDK_ROOT` (unused, should be removed).

File: wii/2_build_toolchain.sh [hunk 2]
```diff
@@ -131,6 +131,8 @@ function install_lib_icu() {
 	cp icudt58l.dat icu-native/source/data/in/
 	cd icu-native/source
 	perl -pi -e 's/SMALL_BUFFER_MAX_SIZE 512/SMALL_BUFFER_MAX_SIZE 2048/' tools/toolutil/pkg_genc.h
+	# glib 2.26 removed xlocale.h: https://ssl.icu-project.org/trac/ticket/13329
```
Comment (carstene1ns) [inline wii/2_build_toolchain.sh]: > glib != glibc :P

**Pull request merged**